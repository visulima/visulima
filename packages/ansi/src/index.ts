export { default as slice } from "./slice";
