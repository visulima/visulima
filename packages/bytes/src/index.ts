import { Buffer } from "node:buffer";

/**
 * Converts a Node.js Buffer to a Uint8Array.
 * Note: While Buffer instances are Uint8Array instances, this function
 * creates a new Uint8Array view on the Buffer's underlying ArrayBuffer,
 * ensuring correct byteOffset and length.
 * @param buf The Buffer to convert.
 * @returns A Uint8Array view of the Buffer.
 */
export const bufferToUint8Array = (buf: Buffer): Uint8Array => new Uint8Array(buf.buffer, buf.byteOffset, buf.length);

/**
 * Checks if a value is a Uint8Array or, in a Node.js environment, a Buffer.
 * @param x The value to check.
 * @returns True if x is a Uint8Array or Buffer, false otherwise.
 */
export const isUint8Array
  = typeof Buffer === "function"
      ? (x: unknown): x is Uint8Array => x instanceof Uint8Array || Buffer.isBuffer(x)
      : (x: unknown): x is Uint8Array => x instanceof Uint8Array;

/**
 * Converts an ASCII string, array of strings, or template literal to a Uint8Array.
 * Each character is converted to its ASCII byte value (0-255). Characters outside this range will be truncated.
 * @param txt The input string, array of strings, or template strings array.
 * @returns A Uint8Array representing the ASCII encoded string.
 */
export const asciiToUint8Array = (txt: TemplateStringsArray | string | [string]): Uint8Array => {
    if (typeof txt === "string")
        return asciiToUint8Array([txt]);

    const [input] = Array.isArray(txt) ? txt : [String.raw(txt as TemplateStringsArray)]; // Handle TemplateStringsArray
    const inputLength = input.length;
    const result = new Uint8Array(inputLength); // Renamed 'res' to 'result'

    for (let index = 0; index < inputLength; index += 1) { // Changed index++ to index += 1
        // eslint-disable-next-line unicorn/prefer-code-point, no-bitwise
        result[index] = input.charCodeAt(index) & 0xFF; // Ensure ASCII range
    }

    return result;
};

/**
 * Converts a UTF-8 string, array of strings, or template literal to a Uint8Array.
 * Requires Node.js Buffer support.
 * @param txt The input string, array of strings, or template strings array.
 * @returns A Uint8Array representing the UTF-8 encoded string.
 */
export const utf8ToUint8Array = (txt: TemplateStringsArray | [string] | string): Uint8Array => {
    if (typeof txt === "string")
        return utf8ToUint8Array([txt]);

    const [input] = Array.isArray(txt) ? txt : [String.raw(txt as TemplateStringsArray)]; // Handle TemplateStringsArray

    return bufferToUint8Array(Buffer.from(input, "utf8"));
};

/**
 * Attempts to convert various data types to a Uint8Array.
 * Supports Uint8Array (returns as is), ArrayBuffer, Array of numbers.
 * In Node.js, also supports Buffer (returns as is or converts from other types via Buffer.from).
 * @param data The data to convert.
 * @returns A Uint8Array representation of the input data.
 * @throws Error 'UINT8ARRAY_INCOMPATIBLE' if the data cannot be converted and Buffer is not available for fallback.
 */
export const toUint8Array = (data: unknown): Uint8Array => {
    if (typeof Buffer === "function" && Buffer.isBuffer(data)) {
        return bufferToUint8Array(data); // Prioritize Buffer and ensure plain Uint8Array
    }

    if (data instanceof Uint8Array) { // This will now handle non-Buffer Uint8Arrays
        return data;
    }

    if (data instanceof ArrayBuffer)
        return new Uint8Array(data);

    // Ensure it's an array of numbers before creating Uint8Array from it
    if (Array.isArray(data) && data.every((item) => typeof item === "number")) {
        return new Uint8Array(data);
    }

    if (typeof Buffer === "function") {
        if (Buffer.isBuffer(data)) {
            return bufferToUint8Array(data); // Use our converter for consistency
        }

        // Attempt to convert string via Buffer.from.
        // Other Buffer-compatible types like ArrayBuffer or ArrayBufferView (which Buffer is)
        // are typically handled by earlier checks or are Buffers themselves.
        if (typeof data === "string") {
            try {
                return bufferToUint8Array(Buffer.from(data));
            } catch {
                // If Buffer.from fails, fall through to error
            }
        }
    }

    throw new Error("UINT8ARRAY_INCOMPATIBLE: Cannot convert data to Uint8Array");
};

// eslint-disable-next-line import/no-extraneous-dependencies
export * from "@std/bytes";
