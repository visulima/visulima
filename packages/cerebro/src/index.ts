// eslint-disable-next-line @typescript-eslint/no-namespace,@typescript-eslint/no-unused-vars
declare namespace NodeJS {
    interface ProcessEnvironment {
        CEREBRO_MIN_NODE_VERSION?: string;
        CEREBRO_OUTPUT_LEVEL: VERBOSITY_LEVEL;
    }
}

export type { Cli } from "./@types/cli";
export type { ArgumentDefinition, Command, OptionDefinition } from "./@types/command";
export type { Extension } from "./@types/extension";
export type { Toolbox } from "./@types/toolbox";

/**
 * Any of the output types [[OUTPUT_NORMAL]], [[OUTPUT_RAW]] and [[OUTPUT_PLAIN]].
 */
export type OutputType = 1 | 2 | 4;

/**
 * Any of the verbosity types
 * [[VERBOSITY_QUIET]], [[VERBOSITY_NORMAL]], [[VERBOSITY_VERBOSE]] and [[VERBOSITY_DEBUG]].
 */
// eslint-disable-next-line @typescript-eslint/naming-convention
export type VERBOSITY_LEVEL = 16 | 32 | 64 | 128 | 256;

export { default } from "./cli";
