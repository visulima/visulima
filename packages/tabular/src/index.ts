export { createGrid, Grid } from "./grid";
export { createTable, Table } from "./table";
export type {
    AutoFlowDirection,
    BorderComponent,
    BorderStyle,
    BorderType,
    CellStyle,
    Content,
    GridCell,
    GridItem,
    GridOptions,
    HorizontalAlignment,
    Style,
    TableCell,
    TableItem,
    TableOptions,
    VerticalAlignment,
} from "./types";
