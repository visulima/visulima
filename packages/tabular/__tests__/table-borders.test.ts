import { blue, gray, green, magenta, red, yellow } from "@visulima/colorize";
import { describe, expect, it } from "vitest";

import { createTable, Table } from "../src";
import { ASCII_BORDER, DEFAULT_BORDER, DOTS_BORDER, DOUBLE_BORDER, MARKDOWN_BORDER, MINIMAL_BORDER, NO_BORDER, ROUNDED_BORDER } from "../src/style";

describe("table borders", () => {
    describe("border styles", () => {
        const sampleData = [
            ["Header 1", "Header 2", "Header 3"],
            ["Row 1 Cell 1", "Row 1 Cell 2", "Row 1 Cell 3"],
            ["Row 2 Cell 1", "Row 2 Cell 2", "Row 2 Cell 3"],
        ];

        it.each([
            ["DEFAULT", DEFAULT_BORDER],
            ["MINIMAL", MINIMAL_BORDER],
            ["DOUBLE", DOUBLE_BORDER],
            ["ROUNDED", ROUNDED_BORDER],
            ["DOTS", DOTS_BORDER],
            ["MARKDOWN", MARKDOWN_BORDER],
            ["ASCII", ASCII_BORDER],
            ["NO_BORDER", NO_BORDER],
        ])("renders table with %s border style", (_, borderStyle) => {
            expect.assertions(1);
            const table = createTable({
                style: {
                    border: borderStyle,
                },
            });

            table.setHeaders(sampleData[0]!);
            table.addRows(...sampleData.slice(1));

            const output = table.toString();
            expect(output).toMatchSnapshot();
        });

        it("should support colored borders", () => {
            expect.assertions(3);

            const table = new Table({
                style: {
                    border: {
                        bodyJoin: { char: gray("│"), width: 1 },
                        bodyLeft: { char: gray("│"), width: 1 },
                        bodyRight: { char: gray("│"), width: 1 },
                        bottomBody: { char: gray("─"), width: 1 },
                        bottomJoin: { char: gray("┴"), width: 1 },
                        bottomLeft: { char: gray("└"), width: 1 },
                        bottomRight: { char: gray("┘"), width: 1 },
                        joinBody: { char: gray("─"), width: 1 },
                        joinJoin: { char: gray("┼"), width: 1 },
                        joinLeft: { char: gray("├"), width: 1 },
                        joinRight: { char: gray("┤"), width: 1 },
                        topBody: { char: gray("─"), width: 1 },
                        topJoin: { char: gray("┬"), width: 1 },
                        topLeft: { char: gray("┌"), width: 1 },
                        topRight: { char: gray("┐"), width: 1 },
                    },
                },
            });

            table.setHeaders(["Header 1", "Header 2"]).addRow([red("Value 1"), "Value 2"]);

            const output = table.toString();

            expect(output).toMatch(/\u001B\[90m.*\u001B\[39m/);
            expect(output).toMatch(/\u001B\[31m.*\u001B\[39m/);
            expect(output).toMatchSnapshot("table-colored-borders");
        });

        it("should support multiple different colors for various border components", () => {
            expect.assertions(6);

            // Based on examples/colorized-border.js
            const table = createTable({
                style: {
                    border: {
                        bodyJoin: { char: blue("│"), width: 1 },
                        bodyLeft: { char: blue("│"), width: 1 },
                        bodyRight: { char: blue("│"), width: 1 },
                        bottomBody: { char: red("─"), width: 1 },
                        bottomJoin: { char: red("┴"), width: 1 },
                        bottomLeft: { char: red("└"), width: 1 },
                        bottomRight: { char: red("┘"), width: 1 },
                        // Assuming headerJoin is handled by joinBody/topJoin etc based on context
                        joinBody: { char: magenta("─"), width: 1 },
                        joinJoin: { char: magenta("┼"), width: 1 },
                        joinLeft: { char: magenta("├"), width: 1 },
                        joinRight: { char: magenta("┤"), width: 1 },
                        topBody: { char: green("─"), width: 1 },
                        topJoin: { char: green("┬"), width: 1 },
                        topLeft: { char: green("┌"), width: 1 },
                        topRight: { char: green("┐"), width: 1 },
                    },
                },
            });

            table
                .setHeaders([
                    { content: "Server", hAlign: "center" },
                    { content: "Load", hAlign: "center" },
                    { content: "Uptime", hAlign: "right" },
                    { content: "Status", hAlign: "center" },
                ])
                .addRow(["API Server", green("28%"), "24d 12h", green("● Online")])
                .addRow(["Database", yellow("78%"), "15d 6h", yellow("● Warning")])
                .addRow(["Cache", red("92%"), "7d 3h", red("● Critical")]);

            const output = table.toString();

            // Basic checks for color codes
            expect(output).toContain("\u001B[34m"); // blue
            expect(output).toContain("\u001B[31m"); // red
            expect(output).toContain("\u001B[35m"); // magenta
            expect(output).toContain("\u001B[32m"); // green
            expect(output).toContain("\u001B[33m"); // yellow

            expect(output).toMatchSnapshot("table-multi-colored-borders");
        });

        it("should support custom border characters", () => {
            expect.assertions(2);

            const table = new Table({
                style: {
                    border: {
                        bodyJoin: { char: "│", width: 1 },
                        bodyLeft: { char: "║", width: 1 },
                        bodyRight: { char: "║", width: 1 },
                        bottomBody: { char: "═", width: 1 },
                        bottomJoin: { char: "╧", width: 1 },
                        bottomLeft: { char: "╚", width: 1 },
                        bottomRight: { char: "╝", width: 1 },
                        joinBody: { char: "─", width: 1 },
                        joinJoin: { char: "┼", width: 1 },
                        joinLeft: { char: "╟", width: 1 },
                        joinRight: { char: "╢", width: 1 },
                        topBody: { char: "═", width: 1 },
                        topJoin: { char: "╤", width: 1 },
                        topLeft: { char: "╔", width: 1 },
                        topRight: { char: "╗", width: 1 },
                    },
                },
            });

            table.addRows(["foo", "bar", "baz"], ["frob", "bar", "quuz"]);

            const expected = ["╔══════╤═════╤══════╗", "║ foo  │ bar │ baz  ║", "╟──────┼─────┼──────╢", "║ frob │ bar │ quuz ║", "╚══════╧═════╧══════╝"].join(
                "\n",
            );

            expect(table.toString()).toBe(expected);
            expect(table.toString()).toMatchSnapshot("table-custom-borders");
        });

        it("should support disabled decoration lines (NO_BORDER specific)", () => {
            expect.assertions(1);
            const table = createTable({ style: { border: NO_BORDER, paddingLeft: 1, paddingRight: 1 } });

            table.addRows(["foo", "bar", "baz"], ["frobnicate", "bar", "quuz"]);

            expect(table.toString()).toMatchSnapshot("table-no-border-specific");
        });

        it("renders table with spanning cells using DEFAULT border style", () => {
            expect.assertions(1);
            const table = createTable({
                style: {
                    border: DEFAULT_BORDER,
                },
            });

            table.setHeaders(["Spanning Header", "Regular Header"]);
            table.addRows([{ colSpan: 2, content: "Spanning Content" }], ["Regular Cell 1", "Regular Cell 2"]);

            const output = table.toString();
            expect(output).toMatchSnapshot();
        });

        it("renders table with multiline content using DOUBLE border style", () => {
            expect.assertions(1);
            const table = createTable({
                style: {
                    border: DOUBLE_BORDER,
                },
            });

            table.setHeaders(["Multiline Header", "Regular Header"]);
            table.addRows(["Line 1\nLine 2\nLine 3", "Regular Cell"], ["Regular Cell 1", "Regular Cell 2"]);

            const output = table.toString();
            expect(output).toMatchSnapshot();
        });

        it("renders table with mixed alignments using ROUNDED border style", () => {
            expect.assertions(1);
            const table = createTable({
                style: {
                    border: ROUNDED_BORDER,
                },
            });

            table.setHeaders([
                { content: "Left Align", hAlign: "left" },
                { content: "Center Align", hAlign: "center" },
                { content: "Right Align", hAlign: "right" },
            ]);
            table.addRows(["Short", "Medium Text", "Longer Text Here"], ["A", "BB", "CCC"]);

            const output = table.toString();
            expect(output).toMatchSnapshot();
        });

        it("renders table with vertical alignments using DOTS border style", () => {
            expect.assertions(1);
            const table = createTable({
                style: {
                    border: DOTS_BORDER,
                },
            });

            table.setHeaders([
                { content: "Top\nAlign", vAlign: "top" },
                { content: "Middle\nAlign", vAlign: "middle" },
                { content: "Bottom\nAlign", vAlign: "bottom" },
            ]);
            table.addRows([
                { content: "Short", vAlign: "top" },
                { content: "Medium\nText", vAlign: "middle" },
                { content: "Long\nText\nHere", vAlign: "bottom" },
            ]);

            const output = table.toString();
            expect(output).toMatchSnapshot();
        });

        it("renders table with word wrapping using MARKDOWN border style", () => {
            expect.assertions(1);
            const table = createTable({
                columnWidths: [8, 13],
                style: {
                    border: MARKDOWN_BORDER,
                },
                wordWrap: true,
            });

            table.setHeaders(["Short Header", "Long Header That Should Wrap"]);
            table.addRows(
                ["Short", "This is a very long cell content that should be wrapped across multiple lines"],
                ["Also Short", "Another long content that needs to be wrapped to maintain readability"],
            );

            const output = table.toString();
            expect(output).toMatchSnapshot();
        });

        it("renders table with truncated content using ASCII border style", () => {
            expect.assertions(1);
            const table = createTable({
                style: {
                    border: ASCII_BORDER,
                },
            });

            table.setHeaders([{ content: "Truncated Header", maxWidth: 10 }, { content: "Regular Header" }]);
            table.addRows(
                [{ content: "This content should be truncated", maxWidth: 10 }, "Regular Cell"],
                [{ content: "Also truncated content here", maxWidth: 10 }, "Another Regular Cell"],
            );

            const output = table.toString();
            expect(output).toMatchSnapshot();
        });

        it("renders table with mixed styling using NO_BORDER style", () => {
            expect.assertions(1);
            const table = createTable({
                style: {
                    border: NO_BORDER,
                    paddingLeft: 2,
                    paddingRight: 2,
                },
            });

            table.setHeaders([
                { content: "Mixed", hAlign: "left", vAlign: "top" },
                { content: "Styling", hAlign: "center", vAlign: "middle" },
                { content: "Example", hAlign: "right", vAlign: "bottom" },
            ]);
            table.addRows([
                { content: "Left\nTop", hAlign: "left", vAlign: "top" },
                { content: "Center\nMiddle", hAlign: "center", vAlign: "middle" },
                { content: "Right\nBottom", hAlign: "right", vAlign: "bottom" },
            ]);

            const output = table.toString();
            expect(output).toMatchSnapshot();
        });
    });

    describe("cell alignment", () => {
        it("should support different cell alignments with borders", () => {
            expect.assertions(8);

            const table = new Table();

            table
                .setHeaders([
                    { content: "Left", hAlign: "left" },
                    { content: "Center", hAlign: "center" },
                    { content: "Right", hAlign: "right" },
                ])
                .addRow([
                    { content: "1", hAlign: "left" },
                    { content: "2", hAlign: "center" },
                    { content: "3", hAlign: "right" },
                ]);

            const output = table.toString();
            const lines = output.split("\n");
            const dataLine = lines.find((line) => line.includes("1"));
            expect(dataLine).toBeDefined();

            const cells = dataLine!.split("│");

            expect(cells[1]).toBeDefined();
            if (cells[1])
expect(cells[1].trimEnd()).toBe(" 1");

            expect(cells[2]).toBeDefined();
            if (cells[2])
expect(cells[2].trim()).toBe("2");

            expect(cells[3]).toBeDefined();
            if (cells[3])
expect(cells[3].trimStart()).toBe("3 ");

            expect(output).toMatchSnapshot("table-alignment-with-borders");
        });
    });

    describe("padding configuration", () => {
        it("should respect padding settings with borders", () => {
            expect.assertions(3);

            const table = new Table({
                style: {
                    paddingLeft: 2,
                    paddingRight: 2,
                },
            });

            table.addRow(["A", "B"]);

            const output = table.toString();
            const lines = output.split("\n");

            const dataLine = lines.find((line) => line.includes("A"))!;
            const cells = dataLine.split("│");

            expect(cells[1]).toBe("  A  ");
            expect(cells[2]).toBe("  B  ");
            expect(output).toMatchSnapshot("table-padding-with-borders");
        });

        it("should handle zero padding", () => {
            expect.assertions(2);

            const table = new Table({
                style: {
                    paddingLeft: 0,
                    paddingRight: 0,
                },
            });

            table.addRow(["A", "B"]);

            const expected = ["┌─┬─┐", "│A│B│", "└─┴─┘"].join("\n");

            expect(table.toString()).toBe(expected);
            expect(table.toString()).toMatchSnapshot("table-zero-padding");
        });
    });
});
