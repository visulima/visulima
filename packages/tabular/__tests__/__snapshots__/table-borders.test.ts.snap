// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`table borders > border styles > renders table with ASCII border style 1`] = `
"+--------------+--------------+--------------+
| Header 1     | Header 2     | Header 3     |
+--------------+--------------+--------------+
| Row 1 Cell 1 | Row 1 Cell 2 | Row 1 Cell 3 |
+--------------+--------------+--------------+
| Row 2 Cell 1 | Row 2 Cell 2 | Row 2 Cell 3 |
+--------------+--------------+--------------+"
`;

exports[`table borders > border styles > renders table with DEFAULT border style 1`] = `
"┌──────────────┬──────────────┬──────────────┐
│ Header 1     │ Header 2     │ Header 3     │
├──────────────┼──────────────┼──────────────┤
│ Row 1 Cell 1 │ Row 1 Cell 2 │ Row 1 Cell 3 │
├──────────────┼──────────────┼──────────────┤
│ Row 2 Cell 1 │ Row 2 Cell 2 │ Row 2 Cell 3 │
└──────────────┴──────────────┴──────────────┘"
`;

exports[`table borders > border styles > renders table with DOTS border style 1`] = `
"┌┈┈┈┈┈┈┈┈┈┈┈┈┈┈┬┈┈┈┈┈┈┈┈┈┈┈┈┈┈┬┈┈┈┈┈┈┈┈┈┈┈┈┈┈┐
┊ Header 1     ┊ Header 2     ┊ Header 3     ┊
├┈┈┈┈┈┈┈┈┈┈┈┈┈┈┼┈┈┈┈┈┈┈┈┈┈┈┈┈┈┼┈┈┈┈┈┈┈┈┈┈┈┈┈┈┤
┊ Row 1 Cell 1 ┊ Row 1 Cell 2 ┊ Row 1 Cell 3 ┊
├┈┈┈┈┈┈┈┈┈┈┈┈┈┈┼┈┈┈┈┈┈┈┈┈┈┈┈┈┈┼┈┈┈┈┈┈┈┈┈┈┈┈┈┈┤
┊ Row 2 Cell 1 ┊ Row 2 Cell 2 ┊ Row 2 Cell 3 ┊
└┈┈┈┈┈┈┈┈┈┈┈┈┈┈┴┈┈┈┈┈┈┈┈┈┈┈┈┈┈┴┈┈┈┈┈┈┈┈┈┈┈┈┈┈┘"
`;

exports[`table borders > border styles > renders table with DOUBLE border style 1`] = `
"╔══════════════╦══════════════╦══════════════╗
║ Header 1     ║ Header 2     ║ Header 3     ║
╠══════════════╬══════════════╬══════════════╣
║ Row 1 Cell 1 ║ Row 1 Cell 2 ║ Row 1 Cell 3 ║
╠══════════════╬══════════════╬══════════════╣
║ Row 2 Cell 1 ║ Row 2 Cell 2 ║ Row 2 Cell 3 ║
╚══════════════╩══════════════╩══════════════╝"
`;

exports[`table borders > border styles > renders table with MARKDOWN border style 1`] = `
"|--------------|--------------|--------------|
| Header 1     | Header 2     | Header 3     |
|--------------|--------------|--------------|
| Row 1 Cell 1 | Row 1 Cell 2 | Row 1 Cell 3 |
|--------------|--------------|--------------|
| Row 2 Cell 1 | Row 2 Cell 2 | Row 2 Cell 3 |
|--------------|--------------|--------------|"
`;

exports[`table borders > border styles > renders table with MINIMAL border style 1`] = `
"----------------------------------------------
  Header 1       Header 2       Header 3      
----------------------------------------------
  Row 1 Cell 1   Row 1 Cell 2   Row 1 Cell 3  
----------------------------------------------
  Row 2 Cell 1   Row 2 Cell 2   Row 2 Cell 3  
----------------------------------------------"
`;

exports[`table borders > border styles > renders table with NO_BORDER border style 1`] = `
" Header 1      Header 2      Header 3     
 Row 1 Cell 1  Row 1 Cell 2  Row 1 Cell 3 
 Row 2 Cell 1  Row 2 Cell 2  Row 2 Cell 3 "
`;

exports[`table borders > border styles > renders table with ROUNDED border style 1`] = `
"╭──────────────┬──────────────┬──────────────╮
│ Header 1     │ Header 2     │ Header 3     │
├──────────────┼──────────────┼──────────────┤
│ Row 1 Cell 1 │ Row 1 Cell 2 │ Row 1 Cell 3 │
├──────────────┼──────────────┼──────────────┤
│ Row 2 Cell 1 │ Row 2 Cell 2 │ Row 2 Cell 3 │
╰──────────────┴──────────────┴──────────────╯"
`;

exports[`table borders > border styles > renders table with mixed alignments using ROUNDED border style 1`] = `
"╭────────────┬──────────────┬──────────────────╮
│ Left Align │ Center Align │      Right Align │
├────────────┼──────────────┼──────────────────┤
│ Short      │ Medium Text  │ Longer Text Here │
├────────────┼──────────────┼──────────────────┤
│ A          │ BB           │ CCC              │
╰────────────┴──────────────┴──────────────────╯"
`;

exports[`table borders > border styles > renders table with mixed styling using NO_BORDER style 1`] = `
"  Mixed    Styling    Example  
  Left     Center       Right  
  Top      Middle      Bottom  "
`;

exports[`table borders > border styles > renders table with multiline content using DOUBLE border style 1`] = `
"╔══════════════════╦════════════════╗
║ Multiline Header ║ Regular Header ║
╠══════════════════╬════════════════╣
║ Line 1           ║ Regular Cell   ║
║ Line 2           ║                ║
║ Line 3           ║                ║
╠══════════════════╬════════════════╣
║ Regular Cell 1   ║ Regular Cell 2 ║
╚══════════════════╩════════════════╝"
`;

exports[`table borders > border styles > renders table with spanning cells using DEFAULT border style 1`] = `
"┌─────────────────┬────────────────┐
│ Spanning Header │ Regular Header │
├─────────────────┴────────────────┤
│ Spanning Content                 │
├─────────────────┬────────────────┤
│ Regular Cell 1  │ Regular Cell 2 │
└─────────────────┴────────────────┘"
`;

exports[`table borders > border styles > renders table with truncated content using ASCII border style 1`] = `
"+------------+----------------------+
| Truncated… | Regular Header       |
+------------+----------------------+
| This cont… | Regular Cell         |
+------------+----------------------+
| Also trun… | Another Regular Cell |
+------------+----------------------+"
`;

exports[`table borders > border styles > renders table with vertical alignments using DOTS border style 1`] = `
"┌┈┈┈┈┈┈┈┬┈┈┈┈┈┈┈┈┬┈┈┈┈┈┈┈┈┐
┊ Top   ┊ Middle ┊ Bottom ┊
┊ Align ┊ Align  ┊ Align  ┊
├┈┈┈┈┈┈┈┼┈┈┈┈┈┈┈┈┼┈┈┈┈┈┈┈┈┤
┊ Short ┊        ┊ Long   ┊
┊       ┊ Medium ┊ Text   ┊
┊       ┊ Text   ┊ Here   ┊
└┈┈┈┈┈┈┈┴┈┈┈┈┈┈┈┈┴┈┈┈┈┈┈┈┈┘"
`;

exports[`table borders > border styles > renders table with word wrapping using MARKDOWN border style 1`] = `
"|--------|-------------|
| Short  | Long Header |
| Header | That Should |
|        | Wrap        |
|--------|-------------|
| Short  | This is a   |
|        | very long   |
|        | cell        |
|        | content     |
|        | that should |
|        | be wrapped  |
|        | across      |
|        | multiple    |
|        | lines       |
|--------|-------------|
| Also   | Another     |
| Short  | long        |
|        | content     |
|        | that needs  |
|        | to be       |
|        | wrapped to  |
|        | maintain    |
|        | readability |
|--------|-------------|"
`;

exports[`table borders > border styles > should support colored borders > table-colored-borders 1`] = `
"[90m┌[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m┬[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m┐[39m
[90m│[39m Header 1 [90m│[39m Header 2 [90m│[39m
[90m├[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m┼[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m┤[39m
[90m│[39m [31mValue 1[39m  [90m│[39m Value 2  [90m│[39m
[90m└[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m┴[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m─[39m[90m┘[39m"
`;

exports[`table borders > border styles > should support custom border characters > table-custom-borders 1`] = `
"╔══════╤═════╤══════╗
║ foo  │ bar │ baz  ║
╟──────┼─────┼──────╢
║ frob │ bar │ quuz ║
╚══════╧═════╧══════╝"
`;

exports[`table borders > border styles > should support disabled decoration lines (NO_BORDER specific) > table-no-border-specific 1`] = `
" foo         bar  baz  
 frobnicate  bar  quuz "
`;

exports[`table borders > border styles > should support multiple different colors for various border components > table-multi-colored-borders 1`] = `
"[32m┌[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m┬[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m┬[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m┬[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m─[39m[32m┐[39m
[34m│[39m   Server   [34m│[39m Load [34m│[39m  Uptime [34m│[39m   Status    [34m│[39m
[35m├[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m┼[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m┼[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m┼[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m┤[39m
[34m│[39m API Server [34m│[39m [32m28%[39m  [34m│[39m 24d 12h [34m│[39m [32m● Online[39m   [34m│[39m
[35m├[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m┼[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m┼[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m┼[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m┤[39m
[34m│[39m Database   [34m│[39m [33m78%[39m  [34m│[39m 15d 6h  [34m│[39m [33m● Warning[39m  [34m│[39m
[35m├[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m┼[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m┼[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m┼[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m─[39m[35m┤[39m
[34m│[39m Cache      [34m│[39m [31m92%[39m  [34m│[39m 7d 3h   [34m│[39m [31m● Critical[39m [34m│[39m
[31m└[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m┴[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m┴[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m┴[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m─[39m[31m┘[39m"
`;

exports[`table borders > cell alignment > should support different cell alignments with borders > table-alignment-with-borders 1`] = `
"┌──────┬────────┬───────┐
│ Left │ Center │ Right │
├──────┼────────┼───────┤
│ 1    │   2    │     3 │
└──────┴────────┴───────┘"
`;

exports[`table borders > padding configuration > should handle zero padding > table-zero-padding 1`] = `
"┌─┬─┐
│A│B│
└─┴─┘"
`;

exports[`table borders > padding configuration > should respect padding settings with borders > table-padding-with-borders 1`] = `
"┌─────┬─────┐
│  A  │  B  │
└─────┴─────┘"
`;
